### PR TITLE
Named Args Unit Tests

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Add unit test and regression for new Named Arguments feature of PythonNet 

#### Related Issue

N/A

#### Motivation and Context

New PythonNet supports KWargs

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Python/NamedArgumentsRegression.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    def Initialize(self):
+        '''Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.'''
+
+        self.SetStartDate(2013,10,8)   #Set Start Date
+        self.SetEndDate(2013,10,17)    #Set End Date
+        self.SetCash(100000)           #Set Strategy Cash
+
+        # Use named args for addition of this security to our algorithm
+        symbol = self.AddEquity(ticker="SPY", resolution=Resolution.Daily).Symbol
```
Comment (Martin-Molinero) [inline Algorithm.Python/NamedArgumentsRegression.py:43]: Let's change the order and assert it still works & would be nice to make **all** calls use named arguments really 🚀

File: Algorithm.Python/NamedArgumentsRegression.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    def OnData(self, data):
+        '''OnData event is the primary entry point for your algorithm. Each new data point will be pumped in here.
+
+        Arguments:
+            data: Slice object keyed by symbol containing the stock data
+        '''
+        if not self.Portfolio.Invested:
+            self.SetHoldings("SPY", 1)
+            self.Debug("Purchased Stock")
```
Comment (Martin-Molinero) [inline Algorithm.Python/NamedArgumentsRegression.py:70]: Missing named arguments I can see:
`for config in self.SubscriptionManager.SubscriptionDataConfigService.GetSubscriptionDataConfigs(symbol):`
`self.SetHoldings("SPY", 1)`
`self.Debug("Purchased Stock")`

unused 
```
from QuantConnect.Indicators import *
import numpy as np
```

File: Tests/Python/PythonNetTests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+using Python.Runtime;
+using QuantConnect.Algorithm;
+using QuantConnect.Research;
+using QuantConnect.Securities.Equity;
+using QuantConnect.Tests.Engine.DataFeeds;
+
+namespace QuantConnect.Tests.Python
+{
+    [TestFixture]
+    public class PythonNetTests
```
Comment (Martin-Molinero) [inline Tests/Python/PythonNetTests.cs:28]: Let's rename the fixture and file to something like NamedArgumentTests, each test should be a different type or named arguments use case I think

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Almost there 👍

**Review by jaredbroad — APPROVED** [review]
Awesome to see named args!!

**Pull request merged**